code/modules/projectiles/gun.dm

```diff
@@ -119,7 +123,7 @@
 		else
 			user.visible_message("<span class='danger'>[user] fires [src]!</span>", "<span class='danger'>You fire [src]!</span>", "You hear a [istype(src, /obj/item/weapon/gun/energy) ? "laser blast" : "gunshot"]!")
 
-	if(heavy_weapon)
+	if(weapon_weight == WEAPON_MEDIUM)
```

Comment:
why == and not >= ? Is there no way weapon_weight could be WEAPON_HEAVY in this situation?